types/resource.go

```diff
@@ -74,25 +83,26 @@ const (
 )
 
 type VolumeSpec struct {
-	Size                    int64          `json:"size,string"`
-	Frontend                VolumeFrontend `json:"frontend"`
-	FromBackup              string         `json:"fromBackup"`
-	NumberOfReplicas        int            `json:"numberOfReplicas"`
-	DataLocality            DataLocality   `json:"dataLocality"`
-	StaleReplicaTimeout     int            `json:"staleReplicaTimeout"`
-	NodeID                  string         `json:"nodeID"`
-	MigrationNodeID         string         `json:"migrationNodeID"`
-	EngineImage             string         `json:"engineImage"`
-	RecurringJobs           []RecurringJob `json:"recurringJobs"`
-	BackingImage            string         `json:"backingImage"`
-	Standby                 bool           `json:"Standby"`
-	DiskSelector            []string       `json:"diskSelector"`
-	NodeSelector            []string       `json:"nodeSelector"`
-	DisableFrontend         bool           `json:"disableFrontend"`
-	RevisionCounterDisabled bool           `json:"revisionCounterDisabled"`
-	LastAttachedBy          string         `json:"lastAttachedBy"`
-	AccessMode              AccessMode     `json:"accessMode"`
-	Migratable              bool           `json:"migratable"`
+	Size                    int64              `json:"size,string"`
+	Frontend                VolumeFrontend     `json:"frontend"`
+	FromBackup              string             `json:"fromBackup"`
+	NumberOfReplicas        int                `json:"numberOfReplicas"`
+	ReplicaAutoBalance      ReplicaAutoBalance `json:"replicaAutoBalance"`
+	DataLocality            DataLocality       `json:"dataLocality"`
+	StaleReplicaTimeout     int                `json:"staleReplicaTimeout"`
```

Comment:
nit: put them in their own section, so the diff is smaller and we have an easier git blame time to see when a field was added.

Reply:
Thanks! fixed.